firebase-config/src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigTest.java

```diff
@@ -1332,6 +1332,23 @@ public void realtime_successfulStatusCode_doesNotIncreaseConfigMetadataBackoffDa
     assertFalse(configRealtimeHttpClientSpy.getBackoffDate().after(backoffDate));
   }
 
+  @Test
+  public void realtime_forbiddenStatusCode_returnsStreamError() throws Exception {
+    ConfigRealtimeHttpClient configRealtimeHttpClientSpy = spy(configRealtimeHttpClient);
+    doReturn(mockHttpURLConnection).when(configRealtimeHttpClientSpy).createRealtimeConnection();
+    doNothing()
+        .when(configRealtimeHttpClientSpy)
+        .closeRealtimeHttpStream(any(HttpURLConnection.class));
+    when(mockHttpURLConnection.getErrorStream())
+        .thenReturn(new ByteArrayInputStream("{ \"code\": 403 }".getBytes(StandardCharsets.UTF_8)));
+    when(mockHttpURLConnection.getResponseCode()).thenReturn(403);
+    configRealtimeHttpClientSpy.beginRealtimeHttpStream();
+
+    verify(configRealtimeHttpClientSpy, never()).startAutoFetch(any());
+    verify(configRealtimeHttpClientSpy, never()).retryHttpConnectionWhenBackoffEnds();
+    verify(mockStreamErrorEventListener).onError(any(FirebaseRemoteConfigServerException.class));
```

Comment:
Maybe add the message we expect from the backend and verify it's parsed correctly?